src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -338,23 +338,26 @@ public CompilerIO(List<string> inputs, List<string> outputs)
         public static CompilerIO GetCompileIO(ProjectContext project, string config, string outputPath, string intermediaryOutputPath, ProjectDependenciesFacade dependencies)
         {
             var compilerIO = new CompilerIO(new List<string>(), new List<string>());
+            var compilationOutput = CompilerUtil.GetCompilationOutput(project.ProjectFile, project.TargetFramework, config, outputPath);
 
             // input: project.json
             compilerIO.Inputs.Add(project.ProjectFile.ProjectFilePath);
 
+            // input: lock file; find when dependencies change
+            compilerIO.Inputs.Add(project.LockFile.LockFilePath);
```

Comment:
Shouldn't a null lock file be a grave restore error, that ought to make everything shutdown before even getting to build?